lib/Models/TerriaJsonCatalogFunction.js

```diff
@@ -443,4 +491,102 @@ TerriaJsonCatalogFunction.prototype.invoke = function() {
   return promise;
 };
 
+// private method split into function to make it recursive
+TerriaJsonCatalogFunction.prototype._handleHttp202 = function(all) {
+    //  A HTTP 202 response to try and get the data from polling acceptedUrl 
+    //    after a delay of acceptedDelay || 10 seconds
+    if (all.status !== undefined && all.status === 202) {
+    // else not a 202 response, all is text from Resource.fetch
+    // and it does not have a status property
+      if (this.acceptedUrl === undefined || 
+        typeof this.acceptedUrl !== "string") {
+          throw new TerriaError({
+            sender: this,
+            title: "Misconfigured init file",
+            message: "Service not configured for TerriaJsonCatalogFunction HTTP 202."
+          });
+      }
+     
+      // JSON-ify the body here, used Resorce.fetch, so body must be text
+      var jsonBody;
+      if (all.response !== undefined && typeof all.response === "string" &&
+          all.response !== "") {
+          jsonBody = JSON.parse(all.response);
+      }
+      if (typeof jsonBody === "object") {
+          all.response = jsonBody;
+      } else {
```

Comment:
We won't include support for non JSON responses for now, so please remove the empty else clause :)

Reply:
I have a re-write that i will commit soon, which involves changing a few more lines around this. into 
try{all.response = JSON.parse(all.response);} catch {}
I have done this because the response body might be empty, which is a syntax error for JSON.parse, and i thought empty clauses were javascript's no-op